src/OpenSage.LowLevel/Graphics2D/DrawingContext.D2D.cs

```diff
@@ -84,16 +100,83 @@ private void PlatformFillRectangle(in RawRectangleF rect, in ColorRgbaF fillColo
             }
         }
 
-        private SolidColorBrush CreateBrush(in ColorRgbaF color)
+        private void PlatformFillTriangle(in RawTriangleF tri, in ColorRgbaF fillColor)
+        {
+            using (var brush = CreateBrush(fillColor))
+            {
+                var geometry = new PathGeometry(_graphicsDevice.DeviceContext.Factory);
+                var sink = geometry.Open();
+                sink.BeginFigure(ToRawVector2(tri.X1, tri.Y1),FigureBegin.Filled);
+                sink.SetFillMode(SharpDX.Direct2D1.FillMode.Winding);
+                sink.AddLine(ToRawVector2(tri.X2, tri.Y2));
+                sink.AddLine(ToRawVector2(tri.X3, tri.Y3));
+                sink.EndFigure(FigureEnd.Closed);
+                sink.Close();
+                _graphicsDevice.DeviceContext.FillGeometry(geometry, brush);
+            }
+        }
+
+        private void PlatformFillTriangle(in RawTriangleF tri, in Texture tex, in RawMatrix3x2 transform)
```

Comment:
Please remove the `in` modifier from the `tex` parameter - it's only useful on `struct` parameters. For reference parameters it's already the default. And please could you use full names, like `texture` instead of `tex`?